src/pymor/parameters/functionals.py

```diff
@@ -74,6 +84,13 @@ def evaluate(self, mu=None):
         else:
             return value
 
+    def d_mu(self, component, index=()):
+        if self.derivative_mappings is None:
+            raise ValueError('You must provide a dict of expressions for the partial derivatives')
+        else:
+            return GenericParameterFunctional(self.derivative_mappings[component][index], self.parameter_type,
+                                              'partial_derivative')
```

Comment:
I would say that `d_mu` should not raise an error when `component` is not contained in `parameter_type`. This code will, however.

Reply:
You mean that it should return zero instead?